src/nvim/option.c

```diff
@@ -3035,6 +3035,8 @@ did_set_string_option (
     // 'signcolumn'
     if (check_opt_strings(*varp, p_scl_values, false) != OK) {
       errmsg = e_invarg;
+    } else {
+      curwin->w_buffer->b_signcols_max = -1;
```

Comment:
Invalidates the cache - could need a comment maybe.  But it is done like this in many places already.

Reply:
probably need `SIGNCOLS_UNKNOWN` in `sign_defs.h` by now.